### PR TITLE
<InputWithOptions/> - fix highlighting for builder options

### PR DESCRIPTION
**Pull request by gendeld**
When rendering an item using `listItemSelectBuilder` inside a `InputWithOptions` with `highlight={true}`,
ListItemSelect is not being displayed

With this PR, items created by builders are shown and highlighted properly.

Before:
https://www.wix-style-react.com/?path=/story/introduction-playground--playground&snippet=69c015a2-02dd-41d0-8ea9-39e0f5a12e09

After:
https://preview.wix-style-react.com/pr_6538/?path=/story/introduction-playground--playground&snippet=69c015a2-02dd-41d0-8ea9-39e0f5a12e09

Screen Shot 2021-01-11 at 17 05 02

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6538

File: src/ListItemSelect/test/ListItemSelect.private.uni.driver.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
 import { listItemSelectDriverFactory as publicDriverFactory } from '../ListItemSelect.uni.driver';
+import { highlighterDriverFactory } from '../../Highlighter/Highlighter.uni.driver';
 
 export const listItemSelectPrivateDriverFactory = base => {
+  const getByDataHook = dataHook => base.$(`[data-hook=${dataHook}]`);
```
Comment (EgoziE) [inline src/ListItemSelect/test/ListItemSelect.private.uni.driver.js:5]: Can it be changed to:
```
import { findByHook } from '../../test/utils/unidriver';
```
?

File: src/ListItemSelect/ListItemSelect.js [hunk 2]
```diff
@@ -110,11 +115,32 @@ class ListItemSelect extends React.PureComponent {
     );
   }
 
+  _renderTitle(textProps) {
+    const { title, optionId = '' } = this.props;
```
Comment (EgoziE) [inline src/ListItemSelect/ListItemSelect.js:119]: Can `optionId = ''` be part of the default props?

**Review by EgoziE — CHANGES_REQUESTED** [review]

**Review by TalZa1810 — COMMENTED** [review]
Few comments

File: src/ListItemSelect/index.d.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ export interface ListItemSelectProps {
   disabled?: boolean;
   highlighted?: boolean;
   onClick?: React.MouseEventHandler<HTMLElement>;
+  optionId?: string | number;
```
[the remark this answers is not in the document]
Comment (gendeld) [reply src/ListItemSelect/index.d.ts:19]: I think I need it for the dataHook on the highlighter

[the remark this answers is not in the document]
Comment (TalZa1810) [reply src/ListItemSelect/index.d.ts:19]: Adding a prop just for a `dataHook`? The `dataHook` is needed for testing something that is visual so I think we can avoid adding this prop and testing the Highlight in a visual test instead.

**Review by EgoziE — APPROVED** [review]

**Review by TalZa1810 — COMMENTED** [review]
few leftovers from the previous change

**Review by TalZa1810 — APPROVED** [review]
Looks good :)
I think that we can add a usage example since we're getting some questions about this feature. We can do it in another PR if you want to merge this first.

File: src/InputWithOptions/InputWithOptions.driver.js [hunk 4]
```diff
@@ -45,13 +45,6 @@ const inputWithOptionsDriverFactory = ({ element, eventTrigger }) => {
     pressKey: key => inputDriver.keyDown(key),
     isMenuOpen: () => popoverTestkit().isContentElementExists(),
     outsideClick: () => popoverTestkit().clickOutside(),
-    isOptionWrappedToHighlighter: optionId => {
```
Comment (TalZa1810) [inline src/InputWithOptions/InputWithOptions.driver.js:48]: For transparency- we saw that this method is not used by our consumers. We've tested this using visual tests and removed this method.

**Pull request merged**